### PR TITLE
Add type definitions for TypeScript

### PR DESCRIPTION
**Pull request by nhunzaker**
I needed type definitions for ars-arsenal and I figured this was the best way to do it!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ dist
 example/public/bundle.js
 example/public/bundle.js.map
 coverage
+node_modules
```
Comment (nhunzaker) [inline .gitignore:5]: I pulled this out into a separate commit on master. Hopefully this goes away soon.

**Review by solomonhawk — APPROVED** [review]
👍

**Review by joeyjoejoejr — APPROVED** [review]
